### PR TITLE
fix: Isolate env variables in `test-connection` command if necessary

### PR DESCRIPTION
**Pull request by disq**
This was missing from the `test-connection` command (we have it in `sync`) and it will be used by the cloud side.
Currently we don't test multiple connections at the same time, but when/if we do, this will come in handy. It also simplifies the handling code as `test-connection` and `sync` behaves the same.

### CODE REVIEW

File: cli/cmd/test_connection.go [hunk 1]
```diff
@@ -112,6 +112,10 @@ func testConnection(cmd *cobra.Command, args []string) error {
 	ctx := cmd.Context()
 	updateSyncTestConnectionStatus(cmd.Context(), log.Logger, cloudquery_api.SyncTestConnectionStatusStarted)
 
+	// in the cloud sync environment, we pass only the relevant environment variables to the plugin
+	isolatePluginEnvironment := apiAuth.NewTokenClient().GetTokenType() == apiAuth.SyncTestConnectionAPIKey
```
Comment (erezrokah) [inline cli/cmd/test_connection.go:116]: There's a fancy `CQ_CLOUD` env you could use, see https://github.com/cloudquery/cloudquery/pull/18128/files#diff-9adce87fe0e31bd984d6a819b5bbbcca2b0f2239249ae2a06f1d502f895fb7adR35

Comment (erezrokah) [reply cli/cmd/test_connection.go:116]: Maybe also update the sync command while you're at it

Comment (disq) [reply cli/cmd/test_connection.go:116]: Unless the `sync` command is already using that, it would be outside of the scope.

Comment (disq) [reply cli/cmd/test_connection.go]: Updated

**Review by erezrokah — APPROVED** [review]

**Comment by disq** [comment]
Merge at will.

**Pull request merged**